docs/spec/lightclient/detection/detection.md

```diff
@@ -363,8 +560,92 @@ The detector shall maintain the following invariants:
 and the following transition invariant
    - *FullNodes' \union Secondaries' \union FaultyNodes' = FullNodes \union Secondaries \union FaultyNodes*
 
+
+> The following invariant is very useful for reasoning, and underlies
+> many intuition when we 
+
+#### **[LCD-INV-TRUSTED-AGREED.1]:**
+
+It is always the case the light client has downloaded a lightblock for height
+*lightStore.LatestTrusted().Height* 
+from each of the current primary and the secondary, that all reported
+the identical lightblock for that height.
+
+> In the above, I guess "the identical" might be replaced with "a
+> matching" to cover commits that might be different.
+
+> The above requires us that before we pick a new secondary, we have to
+> query the secondary for the header of height 
+> *lightStore.LatestTrusted().Height*.
+
+
 ## Solution
 
+### Data Structures
+
+Lightblocks and LightStores are
+defined at [LCV-DATA-LIGHTBLOCK.1] and [LCV-DATA-LIGHTSTORE.1]. See the [verification specification][verification] for details.
+
+> The following data structure [LCV-DATA-POF.1]
+> defines a **proof of fork**. Following
+> [TMBC-SIGN-FORK.1], we require two blocks *b* and *c* for the same
+> height that can both be verified from a common root block *a* (using
+> the skipping or the sequential method).
+
+> [LCV-DATA-POF.1] mirrors the definition [TMBC-SIGN-FORK.1]:
+> *TrustedBlock* corresponds to *a*, and *PrimaryTrace* and *SecondaryTrace*
+> are traces to two blocks *b* and *c*. The traces establish that both
+> *skip-root(a,b,t)* and *skip-root(a,c,t)* are satisfied.
+
+
+#### **[LCV-DATA-POF.1]**:
+```go
+type LightNodeProofOfFork struct {
+    TrustedBlock      LightBlock
+    PrimaryTrace      []LightBlock
+    SecondaryTrace    []LightBlock
+}
+```
+
+
+
+
+
+
+<!-- ```go -->
+<!-- // info about the LC's last trusted block -->
+<!-- type TrustedBlockInfo struct { -->
+<!--   Height              int -->
+<!--   BlockID             BlockID -->
+<!-- }  -->
+<!-- ``` -->
+
+#### **[LCV-DATA-POFSTORE.1]**:
+
+Proofs of Forks are stored in a structure which stores all  proofs
+generated during detection.
+
+```go
+type PoFStore struct {
```

Comment:
We probably don't need this; using array of LightNodeProofOfFork should be sufficient.